File: js/src/types/e2ee.ts
import type { SigningAlgo } from './attestation-common';

/** Model public key and signing algorithm used by the E2EE protocol. */
export type E2eeModelKey = {
  readonly signingAlgo: SigningAlgo;
  /** Hexadecimal model public key, obtained from verified attestation evidence. */
  readonly publicKey: string;
};

export type PrepareE2eeChatRequestParams = {
  readonly request: Request;
  readonly modelKey: E2eeModelKey;
};

/** One encrypted request and the matching response decryption operation. */
export type PreparedE2eeChatRequest = {
  readonly request: Request;
  /** Decrypt JSON or SSE; unsuccessful HTTP responses pass through unchanged. */
  readonly decryptResponse: (response: Response) => Promise<Response>;
};

File: py/src/nearai_inference_sdk/types/e2ee.py
"""Inputs and prepared HTTP messages for standalone Chat E2EE."""

from __future__ import annotations

from collections.abc import Awaitable, Callable
from dataclasses import dataclass

import httpx

from .attestation_common import SigningAlgo


@dataclass(frozen=True, kw_only=True)
class E2eeModelKey:
    """A model public key obtained from verified attestation evidence."""

    signing_algo: SigningAlgo
    public_key: str


@dataclass(frozen=True, kw_only=True)
class PreparedE2eeChatRequest:
    """An encrypted request and the matching JSON/SSE response decryptor."""

    request: httpx.Request
    decrypt_response: Callable[[httpx.Response], Awaitable[httpx.Response]]
